packages/ui-components/src/components/single-select-dropdown/single-select-dropdown.tsx

```diff
@@ -0,0 +1,106 @@
+	/** (optional) If `true` the list is opened */
+	@Prop({ reflect: true }) isOpen?: boolean = false;
+	/** (optional) If `true` the list dropdown is loading */
+	@Prop({ reflect: true }) loading?: boolean = false;
+	/** (optional) The icon to display on the dropdown */
+	@Prop({ reflect: true }) icon?: EIconName | EOtherIconName;
+	/** (optional) If `true` the dropdown is searchable */
+	@Prop({ reflect: true }) searchable?: boolean = false;
+	/** (optional) If `true` dropdown requires a value to be selected */
+	@Prop({ reflect: true }) required?: boolean = false;
+	/** (optional) The text to display on the dropdown label */
+	@Prop({ reflect: true }) label?: string;
+	/** (optional) The text to display on the dropdown  */
+	@Prop({ reflect: true }) value?: string;
+	/** (required) The error state for the dropdown */
+	@Prop({ reflect: true }) errorState?: EValidationState = EValidationState.None;
+	/** (optional) The text to display as help text  */
+	@Prop({ reflect: true }) helpText?: string;
+	/** (optional) If `true` the dropdown is disabled */
+	@Prop({ reflect: true }) disabled?: boolean = false;
+	/** (required) The text to display when there are no options */
+	@Prop({ reflect: true }) noDataAvailableLabel?: string = SINGLE_SELECT_DROPDOWN_NO_DATA_AVAILABLE;
+	/** (optional) The object with the dropdown options */
+	@Prop({ reflect: true }) options?: ISingleSelectDropdownOptions = {};
+	/** (optional) The value of the selected option */
+	@Prop({ reflect: true }) selectedOption?: string;
+
+	/** Emitted when the selected option change */
+	@Event() optionSelected: EventEmitter<string>;
+
+	private selectOption = (event: CustomEvent<string>) => {
+		const selectedOption = event.detail;
+		const option = this.options[selectedOption];
+
+		this.value = option.label;
+		this.optionSelected.emit(option.value);
+		this.isOpen = false;
+	};
+
+	private openStateChangeHandler = (event: CustomEvent<boolean>) => {
+		this.isOpen = event.detail;
+	};
+
+	componentWillLoad() {
+		if (isNil(this.value) && this.selectedOption?.length > 0 && !isEmpty(this.options)) {
+			this.value = this.options[this.selectedOption].label;
+		}
+	}
+
+	@Watch('options')
+	optionsChangeHandler(newValue: { [key: string]: ISingleSelectDropdownOption }) {
+		if (!isNil(this.selectedOption) && !isNil(newValue[this.selectedOption])) {
+			this.value = newValue[this.selectedOption].label;
+		} else {
+			this.value = undefined;
+			this.selectedOption = undefined;
+		}
+	}
+
+	render() {
+		return (
+			<Host>
+				<kv-dropdown
+					isOpen={this.isOpen}
+					label={this.label}
+					value={this.value}
+					loading={this.loading}
+					icon={this.icon}
+					disabled={this.disabled}
+					required={this.required}
+					placeholder={this.placeholder}
+					errorState={this.errorState}
+					helpText={this.helpText}
+					onOpenStateChange={this.openStateChangeHandler}
+				>
+					<kv-dropdown-list searchable={this.searchable}>
```

Comment:
What is the place that we gonna use to access the event `onSearchChange` and rearrange the options by the search?

Reply:
Most dropdown values come from paginated requests, this allows for external search.